gdx/src/com/badlogic/gdx/scenes/scene2d/ui/Spinner.java

```diff
@@ -0,0 +1,262 @@
+	protected void updateValue (boolean increase) {
+		// Update value, and if change occurred, dispatch a change event
+		if ((increase && getModel().next()) || (!increase && getModel().previous())) {
+			layout.setText(style.font, model.getDisplayValue()); // Update layout for new current value
+			ChangeEvent changeEvent = Pools.obtain(ChangeEvent.class);
+			fire(changeEvent);
+			Pools.free(changeEvent);
+		}
+		invalidate();
+	}
+
+	protected SpinnerState getState () {
+		return state;
+	}
+
+	protected void setState (SpinnerState state) {
+		this.state = state;
+	}
+
+	/** Updates the spinner's background and state for the given coordinates
+	 * @param x Current x coordinate of mouse
+	 * @param y Y Current y coordinate of mouse */
+	protected void updateBackgroundForCoords (float x, float y) {
+		float diffX = getWidth() - x;
+		float diffY = y;
+		state = SpinnerState.IDLE;
+		currentBackground = getStyle().background;
+		if (Gdx.input.isButtonPressed(Buttons.LEFT) && diffX < getStyle().background.getRightWidth() && diffX >= 0) {
+			if (diffY <= getHeight() && diffY > getHeight() / 2) {
+				state = SpinnerState.NEXT;
+				currentBackground = getStyle().backgroundNext;
+			} else if (diffY >= 0 && diffY <= getHeight() / 2) {
+				state = SpinnerState.PREVIOUS;
+				currentBackground = getStyle().backgroundPrev;
+			}
+		} else if (diffX >= 0 && diffX <= getWidth() && diffY >= 0 && diffY <= getHeight()) {
+			state = SpinnerState.HOVER;
+			currentBackground = getStyle().backgroundHover;
+		}
+		invalidate();
+	}
+
+	@Override
+	public void draw (Batch batch, float parentAlpha) {
+		currentBackground.draw(batch, getX(), getY(), getWidth(), getHeight());
+		// Center the text
+		style.font.draw(batch, model.getDisplayValue() + "", getX() + style.background.getLeftWidth(),
+			getY() + (getHeight() + layout.height) / 2);
+	}
+
+	@Override
+	public float getPrefWidth () {
+		// Max of borders + text width or drawable width
+		return Math.max(style.background.getMinWidth(),
+			style.background.getLeftWidth() + layout.width + style.background.getRightWidth());
+	}
+
+	@Override
+	public float getPrefHeight () {
+		// Max of borders + text height or drawable height
+		return Math.max(style.background.getMinHeight(),
+			style.background.getTopHeight() + style.background.getBottomHeight() + layout.height);
+	}
+
+	public void setStyle (SpinnerStyle style) {
+		this.style = style;
+		invalidate();
+	}
+
+	public SpinnerStyle getStyle () {
+		return this.style;
+	}
+
+	public void setModel (SpinnerModel model) {
+		this.model = model;
+		invalidate();
+	}
+
+	public SpinnerModel getModel () {
+		return this.model;
+	}
+
+	/** A spinner has four state. When the mouse is not over it, it is IDLE. When the mouse is over it, it is either over the NEXT
+	 * button, the PREVIOUS button or otherwise simply in a HOVER state (mouse is over the widget, but not over the buttons) */
+	enum SpinnerState {
+		IDLE, HOVER, NEXT, PREVIOUS;
```

Comment:
I am not sure on this, but Android, sadly, didn't like enums too much, IIRC. Since this is just internal state, byte constants (C-style) should be fine. Also, it should be possible to have just three states, without the `HOVER`, but I might be mistaken.